packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -161,6 +161,201 @@ void main() {
     expect(renderBox.size.width, 80.0);
   });
 
+  testWidgets('Leading spacing is correct with 0 destinations', (WidgetTester tester) async {
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: null,
+        leading: FloatingActionButton(onPressed: () { }),
+        destinations: const <NavigationRailDestination>[],
+      ),
+    );
+
+    final RenderBox leading = tester.renderObject<RenderBox>(find.byType(FloatingActionButton).at(0));
+    expect(leading.localToGlobal(Offset.zero), const Offset(0, 8.0));
+  });
+
+  testWidgets('Leading spacing is correct with 1 destinations', (WidgetTester tester) async {
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: null,
+        leading: FloatingActionButton(onPressed: () { }),
+        destinations: const <NavigationRailDestination>[
+          NavigationRailDestination(
+            icon: Icon(Icons.favorite_border),
+            selectedIcon: Icon(Icons.favorite),
+            label: Text('Abc'),
+          ),
+        ],
+      ),
+    );
+
+    final RenderBox leading = tester.renderObject<RenderBox>(find.byType(FloatingActionButton).at(0));
+    expect(leading.localToGlobal(Offset.zero), const Offset(12.0, 8.0));
+  });
+
+  testWidgets('Leading spacing is correct with 2 destinations', (WidgetTester tester) async {
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: null,
+        leading: FloatingActionButton(onPressed: () { }),
+        destinations: const <NavigationRailDestination>[
+          NavigationRailDestination(
+            icon: Icon(Icons.favorite_border),
+            selectedIcon: Icon(Icons.favorite),
+            label: Text('Abc'),
+          ),
+          NavigationRailDestination(
+            icon: Icon(Icons.bookmark_border),
+            selectedIcon: Icon(Icons.bookmark),
+            label: Text('Longer Label'),
+          ),
+        ],
+      ),
+    );
+
+    final RenderBox leading = tester.renderObject<RenderBox>(find.byType(FloatingActionButton).at(0));
+    expect(leading.localToGlobal(Offset.zero), const Offset(12.0, 8.0));
+  });
+
+  testWidgets('Trailing spacing is correct with 0 destinations', (WidgetTester tester) async {
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: null,
+        trailing: FloatingActionButton(onPressed: () { }),
+        destinations: const <NavigationRailDestination>[],
+      ),
+    );
+
+    final RenderBox leading = tester.renderObject<RenderBox>(find.byType(FloatingActionButton).at(0));
```

Comment:
trailing

Reply:
Can  you merge the  Trailing spacing with x destination tests with the Leading spacing with with x destination tests?

Reply:
Thanks for the review.
I have corrected the incorrect naming and combined them into one test.

[4c2a3fd](https://github.com/flutter/flutter/pull/152972/commits/4c2a3fd640c5c1e462ec37d836564677ccb05e25)